trollmoves/dispatcher.py

```diff
@@ -405,23 +285,114 @@ def create_dest_url(self, msg, client, conf):
         host_path = urlunsplit((parts.scheme, parts.netloc, path, parts.query, parts.fragment))
         return host_path, connection_parameters, client
 
-    def close(self):
-        """Shutdown the dispatcher."""
-        logger.info('Terminating dispatcher.')
-        self.loop = False
-        try:
-            self.listener.stop()
-        except Exception:
-            logger.exception("Couldn't stop listener.")
-        if self.publisher:
+
+class PublisherReporter:
+    """This class uses a posttroll publisher to report the results of the dispatching.
+
+    This is the first of possibly many reporting classes, eg for monitoring or generating reports daily.
+
+    Idea for future refactoring: This could be made flexible enough to be used in other parts of trollmoves, eg in
+    move_it_client to report moved files. The main problem is the to pass the right configuration/topic for the messages
+    to be correct.
```

Comment:
This also should maybe be in an issue instead of in the docstring.

Reply:
Fixed